kamon-core/src/main/scala/kamon/trace/TracerModule.scala

```diff
@@ -122,12 +122,12 @@ private[kamon] class TracerModuleImpl(metricsExtension: MetricsModule, config: C
 
     val traceToken = token.getOrElse(newToken)
 
-    if (_settings.levelOfDetail == LevelOfDetail.MetricsOnly || !isLocal)
-      newMetricsOnlyContext(traceToken)
-    else {
-      if (!_settings.sampler.shouldTrace)
+    _settings.levelOfDetail match {
+      case LevelOfDetail.MetricsOnly ⇒
         newMetricsOnlyContext(traceToken)
-      else
+      case _ if !isLocal || !_settings.sampler.shouldTrace ⇒
+        newMetricsOnlyContext(traceToken)
+      case _ ⇒
         new TracingContext(traceName, traceToken, true, _settings.levelOfDetail, isLocal, startTimestamp, null, dispatchTracingContext)
```

Comment:
Yeah, I thought it would be better to fix logging problems in another PR, but I didn't know the best way to do it. Maybe a LoggingAdapter could be instantiated when [TracerModule start method](https://github.com/kamon-io/Kamon/blob/master/kamon-core/src/main/scala/kamon/trace/TracerModule.scala#L158) is called ? Or just provide a logging when instantiating TracerModuleImpl like you said.